library/core/src/fmt/mod.rs

```diff
@@ -188,8 +188,26 @@ pub trait Write {
     /// assert_eq!(&buf, "world");
     /// ```
     #[stable(feature = "rust1", since = "1.0.0")]
-    fn write_fmt(mut self: &mut Self, args: Arguments<'_>) -> Result {
-        write(&mut self, args)
+    fn write_fmt(self: &mut Self, args: Arguments<'_>) -> Result {
+        trait SpecWriteFmt {
```

Comment:
Mind adding a comment about how this specialization avoids indirection? Just so it's easier to understand at a glance